src/Std/Time/Zoned/DateTime.lean

```diff
@@ -0,0 +1,350 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+prelude
+import Std.Time.DateTime
+import Std.Time.Zoned.TimeZone
+
+namespace Std
+namespace Time
+
+set_option linter.all true
+
+/--
+It stores a `Timestamp`, a `PlainDateTime` and a `TimeZone`
+-/
+structure DateTime (tz : TimeZone) where
+  private mk ::
+
+  /--
+  `Timestamp` represents the exact moment in time. It's a UTC related `Timestamp`.
+  -/
+  timestamp : Timestamp
+
+  /--
+  `Date` is a `Thunk` containing the `PlainDateTime` that represents the local date and time, it's
+  used for accessing data like `day` and `month` without having to recompute the data everytime.
+  -/
+  date : Thunk PlainDateTime
```

Comment:
I wonder if this would be even better as
```suggestion
  date : Thunk { plain : PlainDateTime // plain = (timestamp.addSeconds tz.toSeconds).toPlainDateTime }
```
with a view towards proofs later.